portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACUtils.macro

```diff
@@ -898,6 +902,24 @@ definition {
 		}
 	}
 
+	macro viewGenericTextInOrder {
+		if (!(isSet(index))) {
+			var index = "1";
+		}
+
+		for (var textValue : list "${textValueList}") {
+			var key_textValue = "${textValue}";
+
+			WaitForPageLoad();
+
+			while (IsElementNotPresent(locator1 = "ACUtils#GENERIC_TEXT_ORDER") && (maxIterations = "50")) {
+				TestUtils.hardRefresh();
+			}
+
+			AssertElementPresent(locator1 = "ACUtils#GENERIC_TEXT_ORDER");
+		}
+	}
+
```

Comment:
Hey Vini, since we are just checking texts, the texts are taking some time to load? I'm seeing that you are using a while loop.

Reply:
No, this macro is based on the one we already have `viewGenericText`, but in this one I created it in case we want to check an order, like when we want to see the most used interests, then we could use this macro. About the loop case, these tests the text doesn't take long to load, but in some case we can use this macro in an area that takes a long time like segments or interests. The macro is not created just thinking about these simple empty state tests

Reply:
Thanks for the clarification!